fileseed.go

```diff
@@ -22,24 +23,26 @@ func NewIndexSeed(dstFile string, srcFile string, index Index) (*FileSeed, error
 		pos:        make(map[ChunkID][]int),
 		index:      index,
 		canReflink: CanClone(dstFile, srcFile),
+		isInvalid:  false,
 	}
 	for i, c := range s.index.Chunks {
 		s.pos[c.ID] = append(s.pos[c.ID], i)
 	}
 	return &s, nil
 }
 
-// LongestMatchWith returns the longest sequence of of chunks anywhere in Source
-// that match b starting at b[0]. If there is no match, it returns nil
+// LongestMatchWith returns the longest sequence of chunks anywhere in Source
+// that match `chunks` starting at chunks[0]. If there is no match, it returns a
+// length of zero and a nil SeedSegment.
 func (s *FileSeed) LongestMatchWith(chunks []IndexChunk) (int, SeedSegment) {
-	if len(chunks) == 0 || len(s.index.Chunks) == 0 {
+	if len(chunks) == 0 || len(s.index.Chunks) == 0 || s.isInvalid {
```

Comment:
Technically, since you're checking the value of `isInvalid` here, you wouldn't actually need the `IsInvalid()` method in the interface anymore. Currently, `IsInvalid()` is called just before this function. Since that check is redundant. Would let you remove that method from selfseed and nullseed as well.

Reply:
Right, I removed `IsInvalid()`.